posts/MarkoB/2017-02-06-validator-service-loader.adoc

```diff
@@ -0,0 +1,310 @@
+Validator provides authors of such libs a possibility to write and share their own validation extensions.
+Which can be picked up by HV and used to validate your data classes.
+
+In this section we will look at http://www.threeten.org/threeten-extra/[ThreeTen Extra types] - a great lib
+that provides additional date-time classes to complement those already present in Java.
+
+As you may already know, Bean Validation (and Hibernate Validator in particular) provides support for
+ https://jcp.org/en/jsr/detail?id=310[JSR 310] date types (javax.time API), by the use of
+`@Past`/`@Future` annotations. So we would want to use these annotations on ThreeTen Extra types as well.
+
+To still keep this example simple we will provide validators only for http://www.threeten.org/threeten-extra/apidocs/org/threeten/extra/YearWeek.html[`YearWeek`]
+and http://www.threeten.org/threeten-extra/apidocs/org/threeten/extra/YearQuarter.html[`YearQuarter`].
+
+We will create a maven project with two modules - one will contain validators and represent a "library" that can be shared,
+another module will be a consumer of this library and will contain some tests.
+You can look at complete code at http://some-valid-github-link-to-hibernate-demo-repo-will-be-here[github].
+Here's a list of dependencies needed for our validators:
+
+[source,xml]
+._pom.xml_
+<dependencyManagement>
+	<dependencies>
+		<dependency>
+			<groupId>javax.validation</groupId>
+			<artifactId>validation-api</artifactId>
+			<version>1.1.0.Final</version>
+		</dependency>
+		<!--HV dependencies-->
+		<dependency>
+			<groupId>org.hibernate</groupId>
+			<artifactId>hibernate-validator</artifactId>
+			<version>5.4.0.Final</version>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.glassfish.web</groupId>
+			<artifactId>javax.el</artifactId>
+			<version>2.2.6</version>
+			<scope>test</scope>
+		</dependency>
+		<!--Date-time model dependencies-->
+		<dependency>
+			<groupId>org.threeten</groupId>
+			<artifactId>threeten-extra</artifactId>
+			<version>1.0</version>
+		</dependency>
+		<!--Test dependencies-->
+		<dependency>
+			<groupId>junit</groupId>
+			<artifactId>junit</artifactId>
+			<version>4.12</version>
+			<scope>test</scope>
+		</dependency>
+	</dependencies>
+</dependencyManagement>
+
+Validator module requires only validation api (`validation-api`) to get `@Past` / `@Future` annotations
+and `ConstraintValidator` interface, which should be implemented by your validator. As for the tests
+we will need Hibernate Validator as a reference implementation of Bean Validation and JUnit as a testing framework.
+
+Where should we start? Well without an implementation of a validator we will not get far. So let's start with
+implementing `ConstraintValidator<Future, YearWeek>` interface which contains two methods:
+
+* `initialize` - initializes validator based on annotation parameters (as of Hibernate Validator 6 this is a default
```

Comment:
The reference to HV 6 can go away, just say it can be empty here as the annotation has no parameters).